### PR TITLE
staceyng | PTBC2-1 | To-Do List - JSON DB

### PR DESCRIPTION
**Pull request by staceyng**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?

Please fill in one error and/or error message you received while working on this assignment.

What part of the assignment did you spend the most time on?

Comfort Level (1-5):

Completeness Level (1-5):

What did you think of this deliverable?

Is there anything in this code that you feel pleased about?

### CODE REVIEW

File: scripts/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      );
+      return;
+    default:
+      const action = args[0];
+      const newItem = args[1] === undefined ? null : args[1];
+      const editedItem = args[2] === undefined ? null : args[2];
+      // process to do list actions
+      toDoListHandler(action, newItem, editedItem);
+  }
+};
```
Comment (upieez) [inline scripts/index.js:23]: in terms of readability here, I would suggest using `if` statements instead

File: scripts/todolist.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { add, read, remove, editOneElement, edit } from "./jsonFileStorage.js";
+
+const filename = "public/data.json";
+
+export const toDoListHandler = (action, newItem, editedItem) => {
+  let idx;
```
Comment (upieez) [inline scripts/todolist.js:6]: at a glance, this variable doesn't give me much context about what it's suppose to store. maybe you can consider renaming it to something more meaningful

File: scripts/index.js (hunk 1 above)
Comment (thereseLYR) [reply scripts/index.js:23]: hi stacey and samuel, could i ask what the `=== undefined ? null : args[1]` achieves here? the syntax looks new to me, and i'd like to learn more.

Comment (upieez) [reply scripts/index.js:23]: hey @thereseLYR, this syntax is the same as an `if else` statement called ternary operator and you can read more about it here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

so at line 18 `args[1] === undefined`, it is checking if `args[1]` is `undefined`.  if it is, return `null` else return `args[1]`

hope that clarifies

**Pull request closed without merging**